### ISSUE TITLE
Uncaught Error: Unexpected value '[object Object]' imported by the module 'AppModule'. Please add a @NgModule annotation.

### ISSUE THREAD

**Issue by BamiGorengo** [jvandemo/generator-angular2-library#124]
Hello,

I follow all steps to "npm link" my dist folder to my angular project. Here is my library, that i made with yogenerator: https://github.com/DevMonkeysDE/kalifa

I import it in my app.modules.ts

```
import {BrowserModule} from '@angular/platform-browser';
import {NgModule} from '@angular/core';
import {BrowserAnimationsModule} from '@angular/platform-browser/animations';

import {AppRoutingModule} from './app-routing.module';
import {AppComponent} from './app.component';
import {
    MdButtonModule, MdCardModule, MdIconModule, MdListModule, MdSidenavModule,
    MdToolbarModule
} from '@angular/material';
import { FrameDefaultComponent } from './frames/frame-default/frame-default.component';
import { SiteHomeComponent } from './sites/site-home/site-home.component';
import { SiteUserprofileComponent } from './sites/site-userprofile/site-userprofile.component';
import { SiteMessengerComponent } from './sites/site-messenger/site-messenger.component';

// Import your library
import { SampleModule } from '@devmonkeys/kalifa';

@NgModule({
    declarations: [
        AppComponent,
        FrameDefaultComponent,
        SiteHomeComponent,
        SiteUserprofileComponent,
        SiteMessengerComponent
    ],
    imports: [
        BrowserModule,
        AppRoutingModule,
        BrowserAnimationsModule,
        SampleModule.forRoot(),
        MdButtonModule,
        MdIconModule,
        MdCardModule,
        MdToolbarModule,
        MdSidenavModule,
        MdListModule
    ],
    providers: [],
    bootstrap: [AppComponent]
})
export class AppModule {
}
```

The Problem is that there is anywhere a bug inside this component. When you check my repository, then you will see, that i 100% dont change anything. Its totaly default build. 

The Readme is not correct. i used a scrope @devmonkeys/kalifa ;)

The errormessage in Chrome:

```
compiler.es5.js:1540 Uncaught Error: Unexpected value '[object Object]' imported by the module 'AppModule'. Please add a @NgModule annotation.
    at syntaxError (http://localhost:4200/vendor.bundle.js:24571:34)
    at http://localhost:4200/vendor.bundle.js:37525:44
    at Array.forEach (native)
    at CompileMetadataResolver.webpackJsonp.../../../compiler/@angular/compiler.es5.js.CompileMetadataResolver.getNgModuleMetadata (http://localhost:4200/vendor.bundle.js:37508:49)
    at JitCompiler.webpackJsonp.../../../compiler/@angular/compiler.es5.js.JitCompiler._loadModules (http://localhost:4200/vendor.bundle.js:48661:66)
    at JitCompiler.webpackJsonp.../../../compiler/@angular/compiler.es5.js.JitCompiler._compileModuleAndComponents (http://localhost:4200/vendor.bundle.js:48620:52)
    at JitCompiler.webpackJsonp.../../../compiler/@angular/compiler.es5.js.JitCompiler.compileModuleAsync (http://localhost:4200/vendor.bundle.js:48582:23)
    at PlatformRef_.webpackJsonp.../../../core/@angular/core.es5.js.PlatformRef_._bootstrapModuleWithZone (http://localhost:4200/vendor.bundle.js:54480:25)
    at PlatformRef_.webpackJsonp.../../../core/@angular/core.es5.js.PlatformRef_.bootstrapModule (http://localhost:4200/vendor.bundle.js:54466:21)
    at Object.../../../../../src/main.ts (http://localhost:4200/main.bundle.js:22047:124)
```

Any idea what i did wrong?

And another Question is: Can i make Librarys, that work like material2? They dont import the whole library as SampleModule.forRoot(), they just import direct the components that they want to use. Like 

```
import {MdButtonModule} from '@angular/material';

@NgModule({
    imports: [
        MdButtonModule
    ],
})
```

This looks for me a little bit cleaner ;)

Hope alot for help! Best Regards!

Sorry, i dont know why i didnt see it... but here is exactly the same issue posted allready: https://github.com/jvandemo/generator-angular2-library/issues/113

But the solution they answere, didnt help me. i get now another error to:

```
client?ffdb:41 [WDS] Errors while compiling. Reload prevented.
client?ffdb:119 ../kalifa/dist/index.js
Module build failed: Error: ENOENT: no such file or directory, open '/Users/budi/PhpstormProjects/kalifa/dist/index.js'
    at Error (native)
 @ ./src/app/app.module.ts 18:0-50
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
```

I just edit my tsconfig.es5.json from:
`"baseUrl": ".",`

to:
```
    "baseUrl": "",
    "paths": {
      "@angular/*": [
        "../node_modules/@angular/*"
      ]
    },
```

**Comment by xxxtonixxx** [comment]
You have to edit the `tsconfig.app.json` of your `angular-cli`, not the `tsconfig.es5.json` file of your lib

From https://github.com/jvandemo/generator-angular2-library/issues/113#issuecomment-305661819

> You need to edit /src/tsconfig.app.json in your project (not library!)...

😉

**Comment by BamiGorengo** [comment]
Dear god.... how i can miss this information so much times... I read each of this this posts 2 ore 3 times :D 

Thanks alot! Works perfectly! ;)

**Comment by xxxtonixxx** [comment]
By order:
- From your index.ts you could export the module that you want, for ex:

Your lib:
```typescript
// Here import to use in NgModule
import { SubModule } from './sub-module';
// Here export to import it when you consume your lib
export { SubModule } from './sub-module';

@NgModule({
...
})
export class SampleModule {
}
```
Your angular-cli app
```typescript
import {AppRoutingModule} from './app-routing.module';
import {AppComponent} from './app.component';

// Import your sub-module
import { SubModule } from '@devmonkeys/kalifa';
// Or better still
import { SubModule } from '@devmonkeys/kalifa/sub-module';

@NgModule({
...
})
export class SampleModule {
}
```
- I don't know exactly why happens that... It should not fail, it was fixed https://github.com/jvandemo/generator-angular2-library/issues/91#issuecomment-304454971 You could try first doing `npm run build:watch` and then start your `angulcar-cli app` (order is important) If you can not resolve it, open a new issue to focus it 😉

**Comment by BamiGorengo** [comment]
Thanks the Order was the Problem ;)

(I ask here, because this is a question, not a issue ;)
But that with the module component import / export makes me nervous... 

This material2 guys have in every module folder multiple components: https://github.com/angular/material2/tree/master/src/lib/datepicker

In every module folder, they place a index.ts where are the components they are included in this module (in my example the datepicker module) 
at last they import and export this index.ts files in the root modules.ts: https://github.com/angular/material2/blob/master/src/lib/module.ts

The problem is that i dont understand the complete code from the material guys... I tryed to reproduce this shema. but i cant make it work...

I want that i include the MessengerModule in my angular App. and can use then all components they are includet in the MessengerModule... Would be verry verry great when you can help me to make this work! ;D

**Comment by jvandemo** [comment]
@BamiGorengo &mdash; I'm going to close this issue because @xxxtonixxx answered/solved your  original question. Please open a new issue if you need help with the module structure so other people can find it by title in the issue list (and potentially jump in to help).

@xxxtonixxx Thanks again for your help, much appreciated! 👍

**Comment by btgoodwin** [comment]
@xxxtonixxx I am running into the same problem with my CLI-based app.  I'm using `@angular/cli` version 1.1.3.  Once I made the above change to the CLI-generated `tsconfig.app.json`, I got past the runtime error and exposed another one.  This time it's the use of `map` from `rxjs/add/operator/map` with the Angular Http service [see here](https://github.com/angular/angular-cli/issues/6788#issuecomment-311690300).  

For me, every problem stems from updating from `@angular/cli` version 1.0.1 to 1.1.3, and yet I'm not finding a lot on what the developers intended you to do with this completely breaking change in a minor point release.
